### ISSUE TITLE
Support plowing effect in 2D

### ISSUE THREAD

**Issue by deepanshubansal01** [osrf/gazebo#3230]
Currently the plowing effect supports surfaces in 1 dimension mainly in longitudinal direction, it would be nice to have plowing effect feature implemented in 2D as well i.e both lateral and longitudinal direction. Further details on plowing effect in longitudinal direction can be found here https://github.com/osrf/gazebo/issues/3085, https://github.com/osrf/gazebo/pull/3164

**Comment by scpeters** [comment]
Are you referring to a lateral plowing effect? #3164 only includes a longitudinal plowing effect

**Comment by deepanshubansal01** [comment]
> Are you referring to a lateral plowing effect? #3164 only includes a longitudinal plowing effect

yes, lateral only sorry for the confusion. I have edited the description.

**Comment by scpeters** [comment]
I think the need for this feature is most obvious when considering the plowing_effect_test world that uses spheres from #3229.

One approach would be use emulate the behavior of the friction cone model and use the wheel linear velocity unit vector as the longitudinal unit vector in the plowing angle calculations instead of deriving it from `fdir1`. This could be an optional parameter.
